### PR TITLE
Persistent storage for comments using Datastore

### PR DESCRIPTION
**Pull request by thompsond**
Step 5 of servlets walkthrough

### CODE REVIEW

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java [hunk 1]
```diff
@@ -20,31 +20,42 @@
...
-    ArrayList<String> messages = new ArrayList<>();
 
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-
-    response.setContentType("application/json;");
+    Query query = new Query("Comment").addSort("time", SortDirection.DESCENDING);
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    PreparedQuery results = datastore.prepare(query);
     String json = "[";
```
Comment (caprichard) [inline portfolio/src/main/java/com/google/sps/servlets/DataServlet.java]: Consider using the StringBuilder class to do appends like these.  It's a lot more efficient than java's string concatenation, which copies the entire string every time you concatenate.

Comment (caprichard) [reply portfolio/src/main/java/com/google/sps/servlets/DataServlet.java]: Also - surely there's some kinda good JSON class in Java (though I don't know for sure what it is).  I feel like there ought to be a class where you can just take an ArrayList (or any Java list, really) and do something like Json.Convert() to it to get the JSON that represents the list of items.  Perhaps ask the tech support folks what they recommend?

**Review by caprichard — CHANGES_REQUESTED** [review]

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java (hunk 1 above)
Comment (thompsond) [reply portfolio/src/main/java/com/google/sps/servlets/DataServlet.java]: Ok I'll check it out

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java [hunk 2]
```diff
@@ -19,34 +19,66 @@
...
-
-    response.setContentType("application/json;");
-    String json = "[";
-    for(String msg : messages) {
-        json += "{\"message\": \"" + msg + "\"},"; 
+    Query query = new Query("Comment").addSort("time", SortDirection.DESCENDING);
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    PreparedQuery results = datastore.prepare(query);
+    Gson gson = new Gson();
+    StringBuilder json = new StringBuilder();
```
Comment (caprichard) [inline portfolio/src/main/java/com/google/sps/servlets/DataServlet.java:42]: It looks like now that you're not doing a whole bunch of appends, you don't need the StringBuilder anymore!

**Review by caprichard — APPROVED** [review]

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java (hunk 2 above)
Comment (thompsond) [reply portfolio/src/main/java/com/google/sps/servlets/DataServlet.java:42]: Ok sounds good!

**Pull request merged**